src/shogun/mathematics/bessel.cpp

```diff
@@ -0,0 +1,290 @@
+// Special functions -*- C++ -*-
+
+// Copyright (C) 2006-2018 Free Software Foundation, Inc.
+//
+// This file is part of the GNU ISO C++ Library.  This library is free
+// software; you can redistribute it and/or modify it under the
+// terms of the GNU General Public License as published by the
```

Comment:
sadly, we cannot do that. It would need to live in shogun-gpl. But let's not add to that. Is there maybe a version in free-domain?

Reply:
(gpl is viral ... it "infects" our bsd repository with gpl and then our repo also must be gpl)

Reply:
OK, I'll keep looking!

Reply:
is the boost license compatible? I'll try and rewrite their code into something that can work standalone

Reply:
It is, but then we would need to include the boost license in our license .... gna

Reply:
> Permission is hereby granted, free of charge, to any person or organization
obtaining a copy of the software and accompanying documentation covered by
this license (the "Software") to use, reproduce, display, distribute,
execute, and transmit the Software, and to prepare derivative works of the
Software, and to permit third-parties to whom the Software is furnished to
do so, all subject to the following:
> The copyright notices in the Software and this entire statement, including
the above license grant, this restriction and the following disclaimer,
must be included in all copies of the Software, in whole or in part, and
all derivative works of the Software, unless such copies or derivative
works are solely in the form of machine-executable object code generated by
a source language processor.

Reply:
I think in that case we should just not have it for libc++. It is quite time consuming to get another implementation, and realistically we will not need it any time soon (this kernel is probably just going to be used for BO)

Reply:
I was looking at how scipy does it, but then we need to add a bunch of fortran code and then translate the scipy C code to something for C++ types. There is a C translation of the main routines but it is but seems very innefficient

Reply:
yeah ... I had similar issues a while ago when using pdf/cdf functions of the Gamma distribution. I eventually found a public domain version of some old-school code, a packet called cdflib `lib/external/cdflib.cpp`

Reply:
it is crazy code, full of goto statements ;)

Reply:
this is what I found https://github.com/cran/Bessel/blob/master/src/zbsubs.c. It's like reading Fortran written in C.. It can't be very efficient with so many jumps :p

Reply:
use it :)

Reply:
OK, I'll have some time during the weekend to clean it up, as this is written to work with Rcpp header files